### PR TITLE
freeciv: 2.4.0 -> 2.5.0

### PR DESCRIPTION
**Pull request by nbp**
Update freeciv version to 2.5.0.
Remove the gtk switch as it is expected to be present by default.

### CODE REVIEW

**Comment by vcunat** [comment]
So it's intentional that the sdl version depends on gtk2 as well?

**Comment by nbp** [comment]
> So it's intentional that now the sdl version depends on gtk2 as well?

Yes, I tried to build without the gtk client, but the configure script does not pass and there is no flag to prevent the lookup for gtk.

**Comment by vbgl** [comment]
The configure setting `--enable-fcmp=cli` disables the GTK interface of an optional component. Both clients gui (GTK3 and SDL) still exist and are different: it seems worth it to keep them both.

**Comment by vcunat** [comment]
Our expression uses gtk2, not gtk3. BTW, the [2.5 news](http://freeciv.wikia.com/wiki/NEWS-2.5.0) claims there's a new Qt interface, but no need for us to add it now.

**Comment by nbp** [comment]
@vbgl , Thanks, I tested and it compiles fine, even with `gtk` disabled.
@vcunat , I noticed, and freeciv is able to compile either against gtk2 or gtk3, so this should be fine.

The Qt interface was already there in the version 2.4.0, and I did not tried it at the time.  I do not recall why, except that I packaged the SDL client because I found the interface nicer, and this is also the one used for the online version.

**Comment by vcunat** [comment]
For myself, the SDL interface style is certainly preferable to the gtk2 one.

File: pkgs/games/freeciv/default.nix [hunk 1]
```diff
@@ -1,32 +1,38 @@
...
-  client = sdlClient || gtkClient;
 
   sdlName = if sdlClient then "-sdl" else "";
   gtkName = if gtkClient then "-gtk" else "";
 
-  baseName = "freeciv-2.4.0";
+  baseName = "freeciv-2.5.0";
 in
 stdenv.mkDerivation {
   name = baseName + sdlName + gtkName;
```
Comment (vcunat) [inline pkgs/games/freeciv/default.nix]: Putting texts *after* version is ugly, I believe. It doesn't work well with our version/update semantics.

Comment (nbp) [inline pkgs/games/freeciv/default.nix]: > Putting texts after version is ugly

I've made a new commit to change that.

**Pull request merged**